datafusion/src/physical_plan/sort_preserving_merge.rs

```diff
@@ -1156,10 +1156,24 @@ mod tests {
     #[tokio::test]
     async fn test_async() {
         let schema = test::aggr_test_schema();
-        let sort = vec![PhysicalSortExpr {
-            expr: col("c7", &schema).unwrap(),
-            options: SortOptions::default(),
-        }];
+        let sort = vec![
```

Comment:
This is required to ensure a consistent sort between the two sorting strategies due to the change to use unstable sorting, introduced in https://github.com/apache/arrow-rs/pull/552. The `SortPreservingMerge` operator happens to be a stable sort but the sort kernel used by the `Sort` Operator no longer is.

The issue here is that the original sort key, `c7` has duplicated values as can be seen in this screenshot (e.g the value `18` is repeated in several rows):
![Screen Shot 2021-09-09 at 1 21 05 PM](https://user-images.githubusercontent.com/490673/132734478-3e57144a-addc-422c-9e96-27b3239ce20e.png)

The full output is in  [basic.txt](https://github.com/apache/arrow-datafusion/files/7138649/basic.txt)
and  [partition.txt](https://github.com/apache/arrow-datafusion/files/7138650/partition.txt)

Reply:
FYI @tustvold

Reply:
@alamb FYI, @yjshen proposed a slightly cleaner fix for this failure by switching to column `c12`: https://github.com/houqp/arrow-datafusion/pull/4

Reply:
Sounds like a good plan if it works 👍

Reply:
Updated this PR to use `c12`